brave/browser-laptop
# History fixes and misc about page fixes

## The big history-fixer-upper
- History code was refactored
  - one step forward with regard to #3502
  - session helper created for aboutHistory (tests added)
  - history specific methods broke out into historyUtil (tests added)
  - history stores it's data in AppStore.about.history (similar structure to new tab code)

- about:history is only updated when something changed
  -  number of calls to render on about:history is drastically reduced (should fix #5382; please reopen if not)
  - entries are added on visit (as expected) and show on about:history instantly
    - Fixes #5542
    - Fixes #4794
    - Fixes #5072

- about:brave text update
Implements followup feedback for #5436 (comment)

Auditors: @cezaraugusto, @darkdh

Test Plan:
1. Launch Brave and go to about:history
2. With history open, open a new tab and visit a site you have never been to before (hint: just append #1234567 to the end of URL)
3. Switch back to history tab and confirm entry shows
4. Open another new tab; in the URL bar, type an entry for a site you HAVE been to before (which shows in the autocomplete)
5. Instead of hitting enter, pick the URL from the autocomplete drop down (use arrow keys and press enter,  OR  click)
6. Visit about:history again and confirm this entry shows
7. On about:history, click some of the entries. They should show their selection status in a faster manner than before

Advanced test notes:
- After exiting Brave, you can view the session file to confirm the details used in about:history and about:brave are removed and NOT stored to disk

## Remove unnecessary/broken/useless context menu items for about pages
Fixes #4812

## Update icon for about pages to fa-list (except for about:newtab)
Fixes #5497

Auditors: @bradleyrichter

Test Plan:
1. Launch Brave and visit about:about
2. Open each of the links and confirm it's the `fa-list` icon


## Screenshots
Here's one showing:
- fixing the casing on component names in about:brave
- new "fa-list" urlbar icon
- context menu has  items removed from about pages
  - save as
   - print
   - view source
   - inspect element

![screen shot 2016-11-11 at 5 13 56 pm](https://cloud.githubusercontent.com/assets/4733304/20233934/55db71e0-a832-11e6-962a-01c23abc3ca9.png)
